packages/ui/src/models/camel/camel-route-resource.test.ts

```diff
@@ -17,3 +19,49 @@ describe('CamelRouteResource', () => {
     expect(resource.getVisualEntities()).toEqual([]);
   });
 });
+
+describe('getCompatibleComponents', () => {
+  it('should not provide isProducerOnly components', () => {
+    const resource = createCamelResource(camelRouteJson);
+    expect(resource.getCompatibleComponents(AddStepMode.ReplaceStep, { path: 'from', label: 'timer' })).toBeDefined;
+  });
+
+  it('should  provide consumerOnly components', () => {
```

Comment:
This statement needs to be reworded a bit differently:

`should NOT provide 'consumerOnly' components`